transport/tchannel/handler.go

```diff
@@ -332,6 +334,15 @@ func getSystemError(err error) error {
 	return tchannel.NewSystemError(tchannelCode, status.Message())
 }
 
+func isTChannelTimeoutError(err error) bool {
+	if tchannelerr, ok := getSystemError(err).(tchannel.SystemError); ok {
```

Comment:
no need to use `getSystemError` -- the error is known to come back from TChannel, so you should use `tchannel.GetSystemErrorCode(err)` and compare that to `ErrCodeTimeout`

this whole function should just be:
```
return tchannel.GetSystemErrorCode(err) == tchannel.ErrCodeTimeout
```

Reply:
Thanks. Amended.